waffle_utils/video_processing/video_tools.py

```diff
@@ -0,0 +1,177 @@
+import os
+from pathlib import Path
+
+import cv2
+
+from waffle_utils.utils import type_validator
+
+
+class VideoTools:
+    def __init__(self, input_path: str, output_path: str):
+        self.input_path = self._input_path = input_path
```

Comment:
In this case, the `_input_path` variable could be accessed from outside the class using instance. it can break encapsulation

Reply:
`self.input_path = self._input_path`
This is for using setter validation.

And `self._input_path = input_path` is encapsulation in python as I know. (but python does not protect it... 😢).

Reply:
> And self._input_path = input_path is encapsulation in python as I know. (but python does not protect it... 😢).

My mistake.
We can encapsule attribute by using two underscore `__` instead of `_`.
I'll apply encapsulation for every waffle code, soon.